start-site/src/test/java/io/spring/start/site/extension/dependency/springmodulith/SpringModulithBuildCustomizerTests.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.spring.start.site.extension.dependency.springmodulith;
+
+import static io.spring.start.site.extension.dependency.springmodulith.SpringModulithBuildCustomizer.*;
+import static org.assertj.core.api.Assertions.*;
+
+import io.spring.initializr.generator.buildsystem.Build;
+import io.spring.initializr.generator.buildsystem.maven.MavenBuild;
+import io.spring.initializr.generator.version.Version;
+import io.spring.initializr.metadata.InitializrMetadata;
+import io.spring.initializr.metadata.support.MetadataBuildItemResolver;
+import io.spring.start.site.extension.AbstractExtensionTests;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Named;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+/**
+ * Tests for {@link SpringModulithBuildCustomizer}.
+ *
+ * @author Oliver Drotbohm
+ */
+class SpringModulithBuildCustomizerTests extends AbstractExtensionTests {
+
+	private final SpringModulithBuildCustomizer customizer = new SpringModulithBuildCustomizer();
+
+	@Test
+	void registersCoreStarterByDefault() {
+
+		Build build = createBuild();
+		customizer.customize(build);
+
+		assertThat(build.dependencies().ids()).contains("modulith");
+	}
+
+	@Test
+	void registersTestStarterByDefault() {
+
+		Build build = createBuild();
+		customizer.customize(build);
+
+		assertThat(build.dependencies().ids()).contains("modulith-starter-test");
+	}
+
+	@Test
+	void registersActuatorStarterIfActuatorsIsPresent() {
+
+		Build build = createBuild();
+		build.dependencies().add("actuator");
+
+		customizer.customize(build);
+
+		assertThat(build.dependencies().ids()).contains("modulith-actuator");
+	}
+
+	@TestFactory
+	Stream<DynamicTest> registersObservabilityStarterIfObservabilityDependencyIsPresent() {
```

Comment:
This is the first use of this in the project. Rather than using a new testing style, I'd like to make sure it can't be achieved with the techniques we're using at the moment.

Reply:
Are you saying you prefer 7 copy & pasted test cases that only differ in the input string? Is `@ParameterizedTest` something that'd be acceptable?

Reply:
Apparently, that is, as I've used it for the test below. Will refactor the test accordingly.

Reply:
> Are you saying you prefer 7 copy & pasted test cases that only differ in the input string

No. I was talking about `DynamicTest`.